sn_interface/src/network_knowledge/section_authority_provider.rs

```diff
@@ -325,5 +350,18 @@ pub mod test_utils {
 
             (signed_sap, nodes, secret_key_set.secret_key())
         }
+
+        pub fn _gen_sk_set<R: RngCore>(
+            rng: &mut R,
```

Comment:
```suggestion
            mut rng: impl RngCore,
```

and in the body, you can take a mutable reference where needed (i.e. &mut rng).

Later on we want to switch `blsttc` to use this `rng: impl RngCore` style API as well.